core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/ScoreOperationChainHandler.java

```diff
@@ -0,0 +1,265 @@
+     * <p>
+     * Defaults to 0.
+     *
+     * @param opAuths a set of operation authorisations
+     * @return maxUserScore the highest score associated with any of the supplied user auths
+     */
+    public Integer getMaxUserAuthScore(final Set<String> opAuths) {
+        Integer maxUserScore = 0;
+        for (final String opAuth : opAuths) {
+            Integer authScore = authScores.get(opAuth);
+            if (null != authScore) {
+                if (authScore > maxUserScore) {
+                    maxUserScore = authScore;
+                }
+            }
+        }
+        LOGGER.info("Returning users max operation chain limit score of {}", maxUserScore);
+        return maxUserScore;
+    }
+
+    protected Integer authorise(final Operation operation) {
+        if (null != operation) {
+            final Class<? extends Operation> opClass = operation.getClass();
+            ArrayList<Class<? extends Operation>> keys = new ArrayList<>(operationScores
+                    .keySet());
+            for (int i = keys.size() - 1; i >= 0; i--) {
+                Class<? extends Operation> key = keys.get(i);
+                if (key.isAssignableFrom(opClass)) {
+                    return operationScores.get(key);
+                }
+            }
+            LOGGER.warn("The operation '{}' was not found in the config file provided the configured default value of {} will be used", operation.getClass().getName(), DEFAULT_OPERATION_SCORE);
+        } else {
+            LOGGER.warn("A Null operation was passed to the OperationChainLimiter graph hook");
+        }
+        return DEFAULT_OPERATION_SCORE;
+    }
+
+    private void loadMapsFromProperties(final String operationScores, final String authScores) {
+        String operationScoresFileName = operationScores;
+        String authScoresFileName = authScores;
+
+        if (operationScoresFileName == null) {
+            operationScoresFileName = (String) System.getProperties().get(OPERATION_SCORES_FILE_KEY);
+        }
+
+        if (authScoresFileName == null) {
+            authScoresFileName = (String) System.getProperties().get(AUTH_SCORES_FILE_KEY);
+        }
+
+        if (authScoresFileName == null || operationScoresFileName == null) {
+            throw new IllegalArgumentException("Auth or operation scores file names not found in system properties");
+        }
+        loadMapsFromProperties(Paths.get(operationScoresFileName), Paths.get(authScoresFileName));
+    }
+
+    private void loadMapsFromProperties(final Path operationScores, final Path authScores) {
+        loadMapsFromProperties(readProperties(operationScores), readProperties(authScores));
+    }
+
+    private void loadMapsFromProperties(final InputStream operationScores, final InputStream authScores) {
+        try {
+            loadMapsFromProperties(readProperties(operationScores), readProperties(authScores));
+        } finally {
+            CloseableUtil.close(operationScores);
```

Comment:
Would a method with varargs be useful or more readable in some cases? e.g.

CloseableUtil.close(operationScores, authScores)